modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1278,17 +1278,15 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 		} else if (member.type == GDScriptParser::ClassNode::Member::VARIABLE && member.variable->property != GDScriptParser::VariableNode::PROP_NONE) {
 			if (member.variable->property == GDScriptParser::VariableNode::PROP_INLINE) {
 				if (member.variable->getter != nullptr) {
-					member.variable->getter->set_datatype(member.variable->datatype);
+					member.variable->getter->return_type = member.variable->datatype_specifier;
+					member.variable->getter->set_datatype(member.get_datatype());
 
 					resolve_function_body(member.variable->getter);
 				}
 				if (member.variable->setter != nullptr) {
-					resolve_function_signature(member.variable->setter);
-
-					if (member.variable->setter->parameters.size() > 0) {
-						member.variable->setter->parameters[0]->datatype_specifier = member.variable->datatype_specifier;
-						member.variable->setter->parameters[0]->set_datatype(member.get_datatype());
-					}
+					ERR_CONTINUE_MSG(member.variable->setter->parameters.is_empty(), "Parser bug: Setter must have an argument.");
```

Comment:
```suggestion
					ERR_CONTINUE_MSG(member.variable->setter->parameters.is_empty(), "Parser bug: Setter must have a parameter.");
```

Or maybe:

```suggestion
					ERR_CONTINUE(member.variable->setter->parameters.is_empty());
```

@AThousandShips What do you think?

Reply:
Argument Vs parameter is a bit mixed, unsure what we use more, but don't know if an explicit error message here is required as a change in this PR

Reply:
`resolve_function_signature()` was removed because we're doing it manually here and because otherwise it will raise a false warning. For getter we don't call it as well.

Yes, "arguments" and "parameters" are synonyms, but I think that we are leaning towards "parameters" when we mean _formal_ arguments. And "arguments" are more often used as _actual_ arguments, passed values of "parameters".

But this is an error that should never happen normally, it probably doesn't make sense to increase the binary size by an useless error message.

Reply:
Agreed on no error message here

Reply:
Removed error message!